pkg/arvo/neo/cod/std/src/fil/diary.md

```diff
@@ -0,0 +1,297 @@
+    =hx-on-submit  "this.reset()"
+    =hx-target     "this"
+    =hx-swap       "afterend"
+    =head          "put-entry"
+    ;date-now(name "id");
+    ;textarea.p2.bd1.br1
+      =name  "text"
+      =placeholder  "today, i ..."
+      =oninput  "this.setAttribute('value', this.value)"
+      =rows  "5"
+      =required  ""
+      =autocomplete  "off"
+      ;
+    ==
+    ;button.p2.b1.br1.bd1.wfc.hover.loader
+      ;span.loaded.s2: create
+      ;span.loading
+        ;+  loading.feather-icons
+      ==
+    ==
+  ==
+++  link-entry
+  |=  [pax=pith =pail:neo]
+  =/  tape  (trip !<(@t q.pail))
+  =/  subject-end  (fall (find [10]~ tape) 56)
+  =/  subject  (scag subject-end tape)
+  =/  id  (trip (snag 0 (pout pax)))
+  ;div.fr.g2
+    ;a.p2.br1.grow.b1.hover.loader
+      =href  "{(en-tape:pith:neo (weld /neo/hawk here.bowl))}/{id}"
+      ;div.loaded.fc.g1.js.as.g2
+        ;span.f3: {(pretty-date `@da`->:pax)}
+        ;span.bold: {subject}
+      ==
+      ;span.loading
+        ;+  loading.feather-icons
+      ==
+    ==
+    ;button.p2.br1.fr.g2.b1.hover.fc.ac.jc.loader
+      =onclick  "alert('not yet implemented. no tombstoning?')"
+      ;span.loaded
+        ;+  close.feather-icons
+      ==
+      ;span.loading
+        ;+  loading.feather-icons
+      ==
+    ==
+  ==
+++  pretty-date
+  |=  date=@da
+  ^-  tape
+  =/  d  (yore date)
+  "{(y-co:co y:d)}-{(y-co:co m:d)}-{(y-co:co d:t:d)}"
+--
+```
+
+There’s only a bit more going on here than in Counter’s `/con/number-htmx.hoon` file. The main thing is that we have `+link-entry` and `+pretty-date` helper arms which are being used like functional frontend components.
+
+The Diary frontend is a text box at the top with a list of entries generated from a list of the `/diary` shrub’s child `pith`s. Nowhere is a list of children in the state being passed from the backend to the frontend, like you might see with a Gall agent passing converting a `(list item)` to a JSON array through a mark in the event that a frontend has scried or subscribed on a path, all of which would require painstaking specification by the developer.
+
+### /con/node-diary-diff.hoon
+```hoon
+/@  node        ::  manx
+/@  diary-diff  ::  ?([%put-entry id=@da txt=@t] [%del-entry id=@da])
+/-  manx-utils
+:-  [%node %$ %diary-diff]
+|=  nod=node
+^-  diary-diff
+=/  mu  ~(. manx-utils nod)
+=/  head  (@tas (got:mu %head))
+=/  id  (slav %da (vol:mu "id"))
+=/  text  (vol:mu "text")
+[%put-entry id text]
+```
+
+Just like the `/con/node-counter-diff.hoon` in the Counter tutorial, all this does is extract attributes from an XML node, having been converted into a `manx`, and manually constructs the poke expected by the shrub.
+
+## Building on Diary
+* Amend the `+kids` arm so that other apps could create shrubs like `/path/to/diary/<@da>/comments`.
+* Diary takes two pokes: `%put-entry` and `%del-entry`, but only one of these is supported on the frontend. Implement a `%dely-entry` poke from the frontend.
```

Comment:
Should be `%del-entry`. Also flagging this as a note to not implement the poke on the frontend ourselves. Could maybe update the alert in the browser `alert('not yet implemented. no tombstoning?')` to prompt the reader to implement it themselves lol

Cc @will-hanlen @hanfel-dovned